internal/pkg/deploy/cloudformation/task_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cloudformation
+
+import (
+    "errors"
+    "testing"
+
+    "github.com/aws/copilot-cli/internal/pkg/deploy"
+
+    "github.com/stretchr/testify/require"
+
+    "github.com/aws/copilot-cli/internal/pkg/aws/cloudformation"
+
+    "github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/mocks"
+    "github.com/golang/mock/gomock"
+)
+
+func TestCloudFormation_DeployTask(t *testing.T) {
+    mockTask := &deploy.CreateTaskResourcesInput{
+        Name: "my-task",
+    }
+
+    testCases := map[string]struct {
+        mockCfnClient func(m *mocks.MockcfnClient)
+        wantedError error
+    }{
+        "create a new stack": {
+            mockCfnClient: func(m *mocks.MockcfnClient) {
+                m.EXPECT().CreateAndWait(gomock.Any()).Return(nil)
```

Comment:
nit: 
```suggestion
                m.EXPECT().CreateAndWait(mockTask).Return(nil)
```

Reply:
The type wouldn't work since `CreateAndWait` accepts `Stack` while `mockTask` is a `CreateTaskResourcesInput`. The later is an input to `StackConfig` and  doesn't implement `Stack`. 

I wonder why do we want specific input to `CreateAndWait`? It accepts a `Stack` type as input, while constructing a stack requires knowledge about `NewTaskStackConfig`.

Reply:
Ooh yeah my bad! This test case is about validating if a new task stack is going to be created correctly, so I thought we should check if the inputs are like what we would expect. The other test cases were fine as they test different behaviors.

I forgot that we need to covert it to a `*cloudformation.Stack` type first though, so it's okay no need to deal with that complexity. 
It'll be cool if in a separate PR we add an integ test for this method though!